src/Stub.php

```diff
@@ -433,26 +438,28 @@ private static function generateMockForAbstractClass(): object
     private static function doGenerateMock($args, $isAbstract = false)
     {
         $testCase = self::extractTestCaseFromArgs($args);
-        $methodName = $isAbstract ? 'getMockForAbstractClass' : 'getMock';
+
+        // PHPUnit 10.4 changed method names
+        if (version_compare(PHPUnitVersion::series(), '10.4', '>=')) {
+            $methodName = $isAbstract ? 'mockObjectForAbstractClass' : 'testDouble';
```

Comment:
https://github.com/sebastianbergmann/phpunit/issues/5241 deprecated `mockObjectForAbstractClass`.
Seems like it will still be available in PHPUnit 11 but will be dropped in PHPUnit 12.